hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMContext.java

```diff
@@ -186,7 +188,9 @@ public long getTermOfLeader() throws NotLeaderException {
 
       if (!isLeader) {
         LOG.warn("getTerm is invoked when not leader.");
-        throw scm.getScmHAManager()
+        StorageContainerManager storageContainerManager =
```

Comment:
Here if it is called from ReconContext it will fail with ClassCastException right?

Reply:
So, should we do this only for when it is StorageContainerManager object lke above

Reply:
Yes we should do this for only StorageContainerManager objects as this operation is specific to it.